databuilder/databuilder/extractor/eventbridge_extractor.py

```diff
@@ -0,0 +1,162 @@
+                            schema.get("description", None),
+                            self._get_property_type(schema),
+                            i,
+                        )
+                    )
+            else:  # NOTE: JSON Schema Draft 4
+                title = content["title"]
+                columns = []
+                for i, (name, schema) in enumerate(content["properties"].items()):
+                    columns.append(
+                        ColumnMetadata(
+                            name,
+                            schema.get("description", None),
+                            self._get_property_type(schema),
+                            i,
+                        )
+                    )
+
+            yield TableMetadata(
+                "eventbridge",
+                "gold",
+                title,
+                registry_name,
+                content.get("description", None),
+                columns,
+            )
+
+    def _get_raw_extract_iter(self, registry_name: str) -> Iterator[Dict[str, Any]]:
+        """
+        Provides iterator of results row from schemas client
+        :return:
+        """
+        schemas_descs = self._search_schemas(registry_name)
+        return iter(schemas_descs)
+
+    def _search_schemas(self, registry_name: str) -> List[Dict[str, Any]]:
+        """
+        Get all schemas descriptions.
+        """
+        paginator = self._schemas.get_paginator("list_schemas")
+        schemas_names = []
+        for result in paginator.paginate(RegistryName=registry_name):
+            for schema in result["Schemas"]:
+                schemas_names.append(schema["SchemaName"])
+
+        schemas_descs = []
+        for schema_name in schemas_names:
+            schema_versions = []
+            data = self._schemas.list_schema_versions(
+                RegistryName=registry_name, SchemaName=schema_name
+            )
+            schema_versions += data["SchemaVersions"]
+            while "NextToken" in data:
+                token = data["NextToken"]
+                data = self._schemas.list_schema_versions(
+                    NextToken=token, RegistryName=registry_name, SchemaName=schema_name
+                )
+                schema_versions += data["SchemaVersions"]
+            latest_schema_version = self._get_latest_schema_version(schema_versions)
+
+            schema_desc = self._schemas.describe_schema(
+                RegistryName=registry_name,
+                SchemaName=schema_name,
+                SchemaVersion=latest_schema_version,
+            )
+
+            schemas_descs.append(schema_desc)
+
+        return schemas_descs
+
+    def _get_latest_schema_version(self, schema_versions: Dict) -> str:
+        versions = []
+        for info in schema_versions:
+            version = int(info["SchemaVersion"])
+            versions.append(version)
+        return str(max(versions))
+
+    def _get_property_type(self, schema: dict) -> str:
```

Comment:
Can we add unit tests for this as well? 
I'd also be interested in some more cases:
- struct inside of a struct (etc)
- struct inside of an array

Reply:
It's already included in the unit tests. I set the samples in a way that tests all the combinations.